src/macaron/policy_engine/policy_engine.py

```diff
@@ -112,3 +114,64 @@ def policy_engine(database_path: str, policy_file: str) -> dict:
             sys.exit(1)
 
         return res
+
+
+def _check_version(database_path: str) -> None:
+    """Verify that database was generated by a compatible version.
+
+    TODO: improve this check and allow version ranges. Perhaps check
+    for major version updates?
+
+    Parameters
+    ----------
+    database_path: str
+        The path to the macaron database
+    """
+    engine = create_engine(f"sqlite:///{database_path}", echo=False)
+
+    with engine.connect() as conn:
+        versions = conn.execute(
+            select(AnalysisTable.macaron_version).where(AnalysisTable.macaron_version != mcn_version)
+        ).scalar()
+        if versions is not None:
+            logger.error("Database generated with unsupported versions (%s).", versions)
+            logger.error("Only databases generated by Macaron version %s are supported.", mcn_version)
+            sys.exit(os.EX_DATAERR)
+
+
+def run_policy_engine(database_path: str, show_prelude: bool, policy_file: str) -> dict:
+    """Evaluate a policy based on configuration and exit.
+
+    Parameters
+    ----------
+    database_path: str
+        The SQLite database file to evaluate the policy against
+    show_prelude: bool
+        Just show the policy prelude and exit.
+    policy_file: str
+        The policy file to evaluate
+
+    Returns
+    -------
+    dict
+        The policy engine result.
```

Comment:
Can you please add some documentation on what are the key and value of the returned dictionary? I think it would be easier to understand what is the internal representation of this dictionary.